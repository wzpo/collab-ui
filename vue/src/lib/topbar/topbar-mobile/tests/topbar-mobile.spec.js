import { mount, shallowMount } from '@vue/test-utils';
import TopbarMobile from '../index.vue';
import Button from '../../../button/index.vue';
import Icon from '../../../icon/index.vue';
import ListSeparator from '../../../list-separator/index.vue';
import Vue from 'vue';

Vue.component(Button.name, Button);
Vue.component(Icon.name, Icon);
Vue.component(ListSeparator.name, ListSeparator);

describe('TopbarMobile', () => {
  it('should match snapshot', () => {
    const container = mount(TopbarMobile, {
      propsData: {
        id: 'test'
      }
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render one TopbarMobile', () => {
    const container = shallowMount(TopbarMobile);

    expect(container.findAll('.md-top-bar__mobile').length).toEqual(1);
  });

  it('should add customized class name if className prop is set', () => {
    const wrapper = shallowMount(TopbarMobile, {
      propsData: {
        class: 'testClassName'
      }
    });

    expect(wrapper.find('.testClassName').exists()).toBeTruthy();
  });

  it('should render children', () => {
    const container = shallowMount(TopbarMobile, {
      slots: {
        default: '<div class="testingforTbM" />'
      }
    });

    expect(
      container.findAll('.testingforTbM').length).toEqual(1);
  });

  it('should handle open', () => {
    const container = mount(
      TopbarMobile
    );

    expect(container.vm.isMobileOpen).toEqual(false);
    container.findAll(Icon).at(0).trigger('click');
    expect(container.vm.isMobileOpen).toEqual(true);
  });

  it('should handle close', () => {
    const container = mount(
      TopbarMobile
    );

    container.vm.isMobileOpen = true;
    expect(container.vm.isMobileOpen).toEqual(true);
    container.find(Icon).trigger('click');
    expect(container.vm.isMobileOpen).toEqual(false);
  });

  it('should not handle keydown close if not space or enter', () => {
    const container = mount(
      TopbarMobile
    );

    container.vm.isMobileOpen = true;
    expect(container.vm.isMobileOpen).toEqual(true);
    container.find('.md-top-bar__mobile').trigger('keydown', { charCode: 40 });
    expect(container.vm.isMobileOpen).toEqual(true);
  });

  it('should handle keydown close on space or enter', () => {
    const container = mount(
      TopbarMobile
    );

    container.vm.isMobileOpen = true;
    expect(container.vm.isMobileOpen).toEqual(true);
    container.find('.md-top-bar__mobile').trigger('keydown', { charCode: 32 });
    expect(container.vm.isMobileOpen).toEqual(false);

    container.vm.isMobileOpen = true;
    expect(container.vm.isMobileOpen).toEqual(true);
    container.find('.md-top-bar__mobile').trigger('keydown', { charCode: 13 });
    expect(container.vm.isMobileOpen).toEqual(false);
  });

  it('should pass down handleClose to children if shouldCloseOnClick is false', () => {
    const container = mount(TopbarMobile, {
      propsData: {
        shouldCloseOnClick: false
      },
      slots: {
        default: '<button class="testButton"/>'
      }
    });

    container.vm.isMobileOpen = true;
    container.find('.testButton').trigger('click');
    expect(container.vm.isMobileOpen).toEqual(false);
  });
});

