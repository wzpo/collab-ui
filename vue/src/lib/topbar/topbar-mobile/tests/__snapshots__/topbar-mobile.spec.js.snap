// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarMobile should match snapshot 1`] = `
<div id="test"><button aria-label="Open Menu" alt="Open Menu" type="button" tabindex="0" role="button" class="md-button md-button--36 md-button--icon md-top-bar__mobile-menu-button"><span class="md-button__children" style="opacity: 1;"><i class="md-icon icon icon-list-menu_20" style="font-size: 20px;"></i></span></button>
  <div role="menu" tabindex="0" class="md-top-bar__mobile md-tb-mobile"><button aria-label="Close Menu" alt="Close Menu" type="button" tabindex="0" role="button" class="md-button md-button--36 md-button--icon md-tb-mobile__close"><span class="md-button__children" style="opacity: 1;"><i class="md-icon icon icon-cancel_20" style="font-size: 20px;"></i></span></button><span></span>
    <div class="md-list-separator"><span class="md-list-separator__container"><!----></span></div>
    <nav class="md-tb-mobile__nav"></nav>
  </div>
  <div role="none" class="md-tb-mobile__mask"></div>
</div>
`;
